u7-components/src/views/BookSearch.vue

```diff
@@ -0,0 +1,60 @@
+<template>
+  <div class="books">
+    <input type="text" v-model="q" placeholder="🔍Search books" />
+
+    <div class="results">
+      <BookCard
+        v-for="book in books"
+        :key="`book-search-${book.isbn13}`"
+        :book="book"
+      />
+    </div>
+  </div>
+</template>
+
+<script lang="ts">
+import { Book, searchBooks } from '@/_services/fakeapi'
+import Vue from 'vue'
+import Component from 'vue-class-component'
+import BookCard from './BookCard.vue'
+@Component({ components: { BookCard } })
+export default class BookSearch extends Vue {
+  q = ''
+  get books (): Book[] {
+    return searchBooks(this.q)
+  }
+
+  created () {
+    this.q = this.$route.query.q as string | null ?? ''
+  }
```

Comment:
น่าสงสัย... ฉันจำไม่ได้ว่าใส่ไว้ในคำแนะนำ!
Suspicious... I don't remember putting this in the instructions!